src/Plot.elm

```diff
@@ -0,0 +1,107 @@
+module Plot where
+
+import Svg exposing (svg, Svg)
+import Private.Extras.SvgAttributes exposing (width, height)
+import Private.Dimensions exposing (Dimensions)
+import Private.Margins as Margins exposing (Margins)
+import Private.BoundingBox as BoundingBox exposing (BoundingBox)
+import Private.Scale.Utils as Scale
+import Private.Scale exposing (Scale)
+import Plot.Axis as Axis exposing (Axis)
+import Private.Bars as Bars exposing (Orient)
+import Private.Axis.View as AxisView
+import Private.Title as Title
+
+type alias Plot =
+  { dimensions: Dimensions
+  , margins: Margins
+  , svgs: List (BoundingBox -> List Svg)
+  , eventHandlers: List (BoundingBox -> Svg.Attribute)
+  , attrs: List Svg.Attribute
+  , title : Title.Model
+  }
+
+createPlot : Float -> Float -> Plot
+createPlot w h =
+  { dimensions = { width = w, height = h }
```

Comment:
Looking through this module, there is no way to update the dimensions without recreating the plot. I can see a future use case where we may want to resize plots dynamically in which case a way to change dimensions would be nice. Since this is not currently needed functionality, punting on the is okay.